packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts

```diff
@@ -111,85 +149,147 @@ class InteractiveGraphQuestionBuilder {
+        startCoords?: CollinearTuple;
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new LinearGraphConfig(options);
         return this;
     }
 
-    withLinear(startCoords?: CollinearTuple): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new LinearConfig(startCoords);
+    withLinearSystem(options?: {
+        coords?: CollinearTuple[];
+        startCoords?: CollinearTuple[];
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new LinearSystemGraphConfig(options);
         return this;
     }
 
-    withLinearSystem(
-        startCoords?: CollinearTuple[],
-    ): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new LinearSystemConfig(startCoords);
+    withRay(options?: {
+        coords?: CollinearTuple;
+        startCoords?: CollinearTuple;
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new RayGraphConfig(options);
         return this;
     }
 
-    withRay(startCoords?: CollinearTuple): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new RayConfig(startCoords);
+    withCircle(options?: {
+        center?: Coord;
+        radius?: number;
+        startCoords?: Coord;
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new CircleGraphConfig(options);
         return this;
     }
 
-    withCircle(startCoords?: Coord): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new CircleGraphConfig(startCoords);
+    withQuadratic(options?: {
+        coords?: [Coord, Coord, Coord];
+        startCoords?: [Coord, Coord, Coord];
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new QuadraticGraphConfig(options);
         return this;
     }
 
-    withQuadratic(
-        startCoords?: [Coord, Coord, Coord],
-    ): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new QuadraticConfig(startCoords);
+    withSinusoid(options?: {
+        coords?: [Coord, Coord];
+        startCoords?: [Coord, Coord];
+    }): InteractiveGraphQuestionBuilder {
+        this.interactiveFigureConfig = new SinusoidGraphConfig(options);
         return this;
     }
 
-    withSinusoid(
-        startCoords?: [Coord, Coord],
+    withPolygon(
+        snapTo: "grid" | "angles" | "sides",
+        options?: {
+            match?: "similar" | "congruent" | "approx";
+            numSides?: number | "unlimited";
+            showAngles?: boolean;
+            showSides?: boolean;
+            coords?: Coord[];
+            startCoords?: Coord[];
+        },
     ): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new SinusoidGraphConfig(startCoords);
+        this.interactiveFigureConfig = new PolygonGraphConfig(snapTo, options);
         return this;
     }
 
-    withPolygon(
-        snapTo: "grid" | "angles" | "sides",
+    withPoint(
```

Comment:
Should this method be named `withPoints`? Since the first argument is `numPoints`, and `withPoints(3)` makes more sense than `withPoint(3)`.